diff --git a/arch/arm/boot/dts/stm32f7-pinctrl.dtsi b/arch/arm/boot/dts/stm32f7-pinctrl.dtsi
index 9314128df185..7ccfcc0bc310 100644
--- a/arch/arm/boot/dts/stm32f7-pinctrl.dtsi
+++ b/arch/arm/boot/dts/stm32f7-pinctrl.dtsi
@@ -284,6 +284,33 @@
 					slew-rate = <2>;
 				};
 			};
+
+			dfsdm_clkout_pins_a: dfsdm-clkout-pins-0 {
+				pins {
+					pinmux = <STM32_PINMUX('D', 3, AF3)>;     /* DFSDM_CKOUT */
+					bias-disable;
+					drive-push-pull;
+					slew-rate = <1>;
+				};
+			};
+
+			dfsdm_clkout_sleep_pins_a: dfsdm-clkout-sleep-pins-0 {
+				pins {
+					pinmux = <STM32_PINMUX('D', 3, ANALOG)>;  /* DFSDM_CKOUT */
+				};
+			};
+
+			dfsdm_data1_pins_a: dfsdm-data1-pins-0 {
+				pins {
+					pinmux = <STM32_PINMUX('C', 3, AF3)>;      /* DFSDM_DATA1 */
+				};
+			};
+
+			dfsdm_data1_sleep_pins_a: dfsdm-data1-sleep-pins-0 {
+				pins {
+					pinmux = <STM32_PINMUX('C', 3, ANALOG)>;   /* DFSDM_DATA1 */
+				};
+			};
 		};
 	};
 };
diff --git a/arch/arm/boot/dts/stm32f746.dtsi b/arch/arm/boot/dts/stm32f746.dtsi
index d6ec55d3adfc..99515999409b 100644
--- a/arch/arm/boot/dts/stm32f746.dtsi
+++ b/arch/arm/boot/dts/stm32f746.dtsi
@@ -122,6 +122,21 @@
 			};
 		};
 
+		sd_adc1: adc-1 {
+			compatible = "sd-modulator";
+			#io-channel-cells = <0>;
+		};
+
+		dfsdm: dfsdm@40017400 {
+			compatible = "st,stm32h7-dfsdm";
+			reg = <0x40017400 0x400>;
+			clocks = <&rcc 1 CLK_DFSDM1>;
+			clock-names = "dfsdm";
+			#interrupt-cells = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+
 		timer2: timer@40000000 {
 			compatible = "st,stm32-timer";
 			reg = <0x40000000 0x400>;
diff --git a/arch/arm/boot/dts/stm32f769-disco.dts b/arch/arm/boot/dts/stm32f769-disco.dts
index f4d8968fedca..0c27def09e42 100644
--- a/arch/arm/boot/dts/stm32f769-disco.dts
+++ b/arch/arm/boot/dts/stm32f769-disco.dts
@@ -187,3 +187,25 @@
 &dma2 {
 	status = "okay";
 };
+
+&dfsdm {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&dfsdm_clkout_pins_a &dfsdm_data1_pins_a>;
+	pinctrl-1 = <&dfsdm_clkout_sleep_pins_a &dfsdm_data1_sleep_pins_a>;
+	spi-max-frequency = <2000000>;
+	status = "okay";
+
+	dfsdm_adc0: filter@0 {
+		compatible = "st,stm32-dfsdm-adc";
+		#io-channel-cells = <1>;
+		reg = <0>;
+		interrupts = <99>;
+		st,adc-channels = <1>;
+		st,adc-channel-names = "in1";
+		st,adc-channel-types = "SPI_R";
+		st,adc-channel-clk-src = "CLKOUT";
+		io-channels = <&sd_adc1>;
+		st,filter-order = <1>;
+		status = "okay";
+	};
+};
\ No newline at end of file
